monkey/monkey_island/cc/next_ui/src/redux/middleware/logout.tsx

```diff
@@ -0,0 +1,17 @@
+import { removeToken } from '@/_lib/authentication';
+import { islandApiSlice } from '@/redux/features/api/islandApiSlice';
+import { Events } from '@/constants/events.constants';
+import { AuthenticationActions } from '@/redux/features/api/authentication/authenticationActions';
+import clearAuthenticationTimer from '@/app/(auth)/_lib/clearAuthenticationTimer';
+
+const logoutMiddleware = (store) => (next) => (action) => {
```

Comment:
What's the reason for using it as a middleware?

Reply:
What are the alternatives?